include/slang/driver/SourceLoader.h

```diff
@@ -152,6 +159,64 @@ class SLANG_EXPORT SourceLoader {
     /// it does not exist. Returns nullptr if @a name is empty.
     SourceLibrary* getOrAddLibrary(std::string_view name);
 
+    /// Find a source buffer by searching through libraries added via -y
+    SourceBuffer findBuffer(std::string_view name);
+
+    /// Load trees using a custom buffer finder function (templated with concept constraint)
+    template<BufferFinder FindBufferFunc>
+    static void loadTrees(
```

Comment:
I think I'd prefer that you use a `function_ref` argument instead of templating on the callback so that the implementation can remain hidden. I think the tiny perf overhead here doesn't matter.